### PR TITLE
perf(admin-setting): optimize loading of options #2491

### LINKED ISSUE #2491: perf(admin-setting): optimize loading of options
## Issue Overview
If you run the following SQL query:

`SELECT * FROM wp_options WHERE option_name LIKE '%give_%' AND autoload='yes'`

… Then you will find that there is certain data that we are storing in option table that has autoload set to true. This means that it will load on each page even we do not need it:

image_1

**The `get_option()` function internally caches all autoload options, so Give options will load only the first time.**

### The Issue with Autoload

* It adds unwanted weight for first the get all options query.

* We are storing all core settings in a serialize array. So every time we have to get all core setting even if we only need:

    * Page related setting

    * Currency related setting

    * Email setting

    * Advance setting etc.

## Solution

To resolve above issue we can do following:

* Stop setting autoload to true for all core options.

* Group similar settings into their own groups. For example:

    * Currency settings

    * Page settings

    * Payment gateway setting ( per payment gateway )

    * Email setting

To achieve this we have to do following this:

1. Develop a Core setting upgrade routine:
This will restore setting according to their `setting_page[section][tab][setting_name1]`, `setting_page[section][tab][setting_name2]`, etc.

2. Update admin core render and save logic with backward compatibility:

image_2

3. Update `give_get_option` to get setting by each section, tab or all with backward compatibility

#### Related Links:

* [https://wordpress.stackexchange.com/a/83776/29439](https://wordpress.stackexchange.com/a/83776/29439)

### PR DESCRIPTION
**Pull request by ravinderk**
## Description
This PR will resolve #2491

## How Has This Been Tested?
Running updates for `2.2.0`

## Screenshots (jpeg or gifs if applicable):
image

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/upgrades/upgrade-functions.php [hunk 1]
```diff
@@ -2660,6 +2664,65 @@ function give_v220_upgrades(){
 	give_v220_delete_wp_session_data();
 }
 
+/**
+ * Version 2.3.0 automatic updates
+ *
+ * @since 2.3.0
+ */
+function give_v230_upgrades(){
```
Comment (DevinWalker) [inline includes/admin/upgrades/upgrade-functions.php]: @ravinderk update function name

Comment (ravinderk) [reply includes/admin/upgrades/upgrade-functions.php]: @DevinWalker thank you for catching that

**Comment by ravinderk** [comment]
@DevinWalker Changes implemented

**Comment by raftaar1191** [comment]
Working perfectly try with DB update also by update the Give Settings

**Pull request merged**